src/test_utils.jl

```diff
@@ -173,6 +313,107 @@ function test_interface(
     @test all(length.(predictions) .== size(outputs, 1))
 end
 
+function test_interface(
+    template::Template, ::Type{PointEstimate}, ::Type{SingleOutput}, ::Type{DistributionPredictInput};
+    inputs=[MvNormal(5, m) for m in 1:5], outputs=rand(1, 5),
+)
+    predictions = test_common(template, inputs, outputs)
+    @test predictions isa NamedDimsArray{(:variates, :observations), <:Real, 2}
+    @test size(predictions) == size(outputs)
+    @test size(predictions, 1) == 1
+end
+
+function test_interface(
```

Comment:
the number of methods we need to add here worries me.

can we somehow have 
```
function test_interface(template::Template, kwargs...)
    test_common(template; kwargs...)
    test_estimate_type(estimate_type(template), template; kwargs...)
    test_output_type(output_type(template), template; kwargs...)
    test_predict_input_type(predict_input_type(template), template; kwargs...)
end
```
then if a new trait is added, we just define
```
test_my_new_trait(::NewTraitSubtype, ::Template; kwargs...)
```
?

Reply:
I wondered if we'd hit the point of needing a refactor.  I think something like the above should work, I'll try it out.

Do you think it's alright to do this refactor in the same PR as adding the new trait?